recipes/cmake/3.x.x/conanfile.py

```diff
@@ -34,7 +36,7 @@ def _configure_cmake(self):
             self._cmake = CMake(self)
             self._cmake.definitions["CMAKE_BOOTSTRAP"] = False
             if self.settings.os == "Linux":
-                self._cmake.definitions["OPENSSL_USE_STATIC_LIBS"] = True
+                self._cmake.definitions["OPENSSL_USE_STATIC_LIBS"] = not self.options["openssl"].shared
                 self._cmake.definitions["CMAKE_EXE_LINKER_FLAGS"] = "-lz"
```

Comment:
Does this mean we need to add zlib as a dep, too?

Reply:
I think it should, but I did the minimum amount of changes because it seemed to work on CI.
What does @Croydon have to say about this?

Reply:
Does cmake requires zlib? Otherwise no. OpenSSL 1.1.x doesn't require zlib anymore.

Reply:
```suggestion
```